webpack/banner_size_warning_plugin.js

```diff
@@ -0,0 +1,60 @@
+const chalk = require( 'chalk' ).default;
+
+class BannerSizeWarningPlugin {
+	constructor( max_banner_size ) {
+		this.banner_size = max_banner_size.limit; // limit in bytes
+	}
+
+	apply( compiler ) {
+		compiler.hooks.done.tap(
+			{ name: 'BannerSizeWarningPlugin', stage: Infinity },
+			( stats ) => {
+				const banners = [];
+
+				for ( const bannerName in stats.compilation.assets ) {
+					const banner = stats.compilation.assets[ bannerName ];
+					const size = banner.size();
+					if ( size > this.banner_size ) {
+						banners.push( { name: bannerName, size } );
+					}
+				}
+
+				if ( banners.length > 0 ) {
+					setTimeout( () => {
+						console.warn(
+							chalk.yellowBright( `\nWarning - The following banners exceed the size limit of ` ) +
+							chalk.green( `${ this.banner_size / 1024 }KB` ) +
+							chalk.yellowBright( `:\n` )
+						);
+
+						banners.forEach( banner =>
+							console.warn( `- ${ banner.name }:`,
+								chalk.red( `${ ( banner.size / 1024 ).toFixed( 2 ) }KB` ) )
+						);
+
+						console.warn(
+							chalk.yellowBright( `\nPlease do following two things:\n\n`,
+								`1. Notify the FUN team and consider removing or optimizing features.\n`,
+								`2. Increase the max banner size limit by 20KB (in 'webpack.production.js' file)`
+						) );
+
+						console.warn(
+							chalk.greenBright( `\n\nExample Email format:` ),
+							`\n\nDear FUN team, \nThe following banners exceed the banner size limit`,
+							`(${ this.banner_size / 1024 }KB) after compilation:\n` );
+
+						banners.forEach( banner =>
+							console.warn( `- ${ banner.name }:`,
+								chalk.red( `${ ( banner.size / 1024 ).toFixed( 2 ) }KB` ) )
+						);
+
+						console.warn( `\nFollowing features make it happen:\n---Features list---`,
+						`\n\nCould we please remove or optimise this feature?` );
```

Comment:
I like it, it's very precise :+1: 

Except the last sentence, where I think they probably can't answer this question to us very well (they don't know how to technically optimise, they rather would give up old features than the new current one, ...)
Maybe we could have a plea like 
"Please consider scraping older banner features or reducing the scope of the new feature."